bears/general/IndentationBear.py

```diff
@@ -0,0 +1,216 @@
+        """
+        ranges = []
+        for open_indent in indent_types:
+            close_indent = indent_types[open_indent]
+            open_pos = list(self.get_valid_sequences(
+                                     file, open_indent, annotation_dict))
+            close_pos = list(self.get_valid_sequences(
+                                     file, close_indent, annotation_dict))
+
+            to_match = len(open_pos) - 1
+            while to_match >= 0:
+                close_index = 0
+                while close_index < len(close_pos):
+                    if(open_pos[to_match].position
+                       <= close_pos[close_index].position):
+                        ranges.append(
+                            SourceRange.from_absolute_position(
+                                            filename,
+                                            open_pos[to_match],
+                                            close_pos[close_index]))
+                        close_pos.remove(close_pos[close_index])
+                        open_pos.remove(open_pos[to_match])
+                        to_match -= 1
+                        break
+                    close_index += 1
+                if((len(close_pos) == 0 and to_match != -1) or
+                   (len(close_pos) != 0 and to_match == -1)):
+                    # None to specify unmatched indents
+                    raise UnmatchedIndentError(open_indent, close_indent)
+
+        # Ranges are returned in the order of least nested to most nested
+        # and also on the basis of which come first
+        return tuple(ranges)[::-1]
+
+    @staticmethod
+    def get_valid_sequences(file, sequence, annotation_dict):
+        """
+        A vaild sequence is a sequence that is outside of comments or strings.
+
+        :param file:            File that needs to be checked in the form of
+                                a list of strings.
+        :param sequence:        sequence whose validity is to be checked.
+        :param annotation_dict: A dictionary containing sourceranges of all the
+                                strings and comments within a file.
+        :return:                A tuple of AbsolutePosition's of all occurances
+                                of sequence outside of string's and comments.
+        """
+        file_string = ''.join(file)
+        # tuple since order is important
+        sequence_positions = tuple()
+        for sequence_match in unescaped_search_for(sequence, file_string):
+            valid = True
+            sequence_position = AbsolutePosition(
+                                    file, sequence_match.start())
+            # ignore if within string
+            for string in annotation_dict['strings']:
+                if(sequence_position >= string.start and
+                   sequence_position <= string.end):
+                    valid = False
+            # ignore if within comments
```

Comment:
generally, don't be afraid of newlines ;)

they are in no way required but personally I would put blank lines in front of lines 187, 191 and 196. Just look though your code where else they look nice and try to adapt it as a habit :)

It makes the code easier to read and nobody cares about number of lines of a module.